pkg/controller/core.oam.dev/v1alpha2/application/revision.go

```diff
@@ -20,97 +20,50 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog/v2"
 	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/oam-dev/kubevela/apis/core.oam.dev/common"
-	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
 	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	"github.com/oam-dev/kubevela/pkg/appfile"
 	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/dispatch"
 	"github.com/oam-dev/kubevela/pkg/controller/utils"
 	"github.com/oam-dev/kubevela/pkg/oam"
 	"github.com/oam-dev/kubevela/pkg/oam/util"
 )
 
-// AppRevisionHash is used to compute the hash value of the AppRevision
-type AppRevisionHash struct {
-	ApplicationSpecHash     string
-	WorkloadDefinitionHash  map[string]string
-	ComponentDefinitionHash map[string]string
-	TraitDefinitionHash     map[string]string
-	ScopeDefinitionHash     map[string]string
-}
-
-// UpdateRevisionStatus will update the status of Application object mainly for update the revision part
-func (h *appHandler) UpdateRevisionStatus(ctx context.Context, revName, hash string, revision int64) error {
-	h.app.Status.LatestRevision = &common.Revision{
-		Name:         revName,
-		Revision:     revision,
-		RevisionHash: hash,
-	}
-	// make sure that we persist the latest revision first
-	if err := h.r.UpdateStatus(ctx, h.app); err != nil {
-		klog.InfoS("Failed to update the latest appConfig revision to status", "application", klog.KObj(h.app),
-			"latest revision", revName, "err", err)
-		return err
-	}
-	klog.InfoS("Recorded the latest appConfig revision", "application", klog.KObj(h.app),
-		"latest revision", revName)
-	return nil
-}
-
-// setRevisionMetadata will set the ApplicationRevision with the same annotation/label as the app
-func (h *appHandler) setRevisionMetadata(appRev *v1beta1.ApplicationRevision) {
-	appRev.Namespace = h.app.Namespace
-	appRev.SetAnnotations(h.app.GetAnnotations())
-	appRev.SetLabels(h.app.GetLabels())
-	util.AddLabels(appRev, map[string]string{oam.LabelAppRevisionHash: h.revisionHash})
-	appRev.SetOwnerReferences([]metav1.OwnerReference{{
-		APIVersion: v1beta1.SchemeGroupVersion.String(),
-		Kind:       v1beta1.ApplicationKind,
-		Name:       h.app.Name,
-		UID:        h.app.UID,
-		Controller: pointer.BoolPtr(false),
-	}})
-}
-
-// setRevisionWithRenderedResult will set the ApplicationRevision with the rendered result
-// it's ApplicationConfiguration and Component for now
-func (h *appHandler) setRevisionWithRenderedResult(appRev *v1beta1.ApplicationRevision, ac *v1alpha2.ApplicationConfiguration,
-	comps []*v1alpha2.Component) {
-	appRev.Spec.Components = ConvertComponents2RawRevisions(comps)
-	appRev.Spec.ApplicationConfiguration = util.Object2RawExtension(ac)
-	appRev.Spec.ResourcesConfigMap.Name = appRev.Name
-}
```

Comment:
why all these above not used?

Reply:
I made a conversion between AppConfig&Components and ComponentManifests. This logic is kept now.